app/src/test/java/calculator/example/com/kotlincalculator/mvp/CalculatorPresenterTest.kt

```diff
@@ -0,0 +1,124 @@
+package calculator.example.com.kotlincalculator.mvp
+
+import calculator.example.com.kotlincalculator.CalculatorActivity
+import calculator.example.com.kotlincalculator.bus.observer.OnCalculatorActionButtonPressedBusObserver
+import calculator.example.com.kotlincalculator.bus.observer.OnCalculatorClearButtonPressedBusObserver
+import calculator.example.com.kotlincalculator.bus.observer.OnCalculatorEqualButtonPressedBusObserver
+import calculator.example.com.kotlincalculator.bus.observer.OnCalculatorNumberButtonPressedBusObserver
+import calculator.example.com.kotlincalculator.mvp.model.CalculatorActions
+import calculator.example.com.kotlincalculator.mvp.model.CalculatorModel
+import calculator.example.com.kotlincalculator.mvp.view.CalculatorView
+import com.globant.counter.utils.bus.RxBus
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mock
+import org.mockito.Mockito.*
+import org.mockito.MockitoAnnotations
+import org.mockito.Mockito.`when` as whenever
+
+class CalculatorPresenterTest {
+
+    private var presenter: CalculatorPresenter? = null
+    private var model: CalculatorModel? = null
```

Comment:
you could just mock the model as well, because you want to test the presenter logic

Reply:
I could, however in this test Im also testing the interactions between the `Presenter` and the `model ` if I mock the `model` i would have to mock every interaction (by using mockito for example) and as in this case the model is not a complex object (so the effort to do it does not make sense neither) I will keep it as non-mock so it will operate normally